internal/client/multi.go

```diff
@@ -18,6 +21,6 @@ func (clients multiClient) ExpireTime(domain string) (time.Time, error) {
 
 // NewMultiClient returns a client that wraps multiple clients.
 // It returns the first success, or, if all clients fail, the latest failure.
-func NewMultiClient(clients ...Client) Client {
+func NewMultiClient(clients ...Client) multiClient {
```

Comment:
since `multiClient` is not exported... I'm not sure about this... maybe its better to leave it as just `Client` as before or export `multiClient`...

Reply:
It's recommended way to return structs instead of interfaces: https://github.com/golang/go/wiki/CodeReviewComments#interfaces
 "accept interfaces, return structs" principle
 It's okay to return private struct. If struct implement other interfaces, it has to return struct only.

Reply:
its not the same thing, in their examples the type is exported... also, the interfaces are not defined for mocking, although they are used for this purpose as well... so I'm not sure that applies here...

Reply:
private struct will be visible via interfaces (case, which we need).
If you wish to mock argument, it will be easy to do using interface

Reply:
so, it's not so important change and pretty discussable, i will return code back)